modules/test/playwright/tests/style-book-web/styleBookDisplayPageTemplatePreview.spec.ts

```diff
@@ -0,0 +1,247 @@
+
+				return computedStyle.color;
+			});
+
+			expect(fragmentColor).toBe(colorRGB);
+		};
+
+		await page.getByRole('button', {name: 'Color System'}).click();
+
+		await page.getByRole('menuitem', {name: 'General'}).click();
+
+		await selectOptionAndUpdateColor(
+			displayPageTemplateNameBlogsEntry,
+			displayPageTemplateNameBlogsEntry,
+			'#995511',
+			'rgb(153, 85, 17)'
+		);
+
+		await selectOptionAndUpdateColor(
+			displayPageTemplateNameBlogsEntry,
+			displayPageTemplateNameWebContent,
+			'#666666',
+			'rgb(102, 102, 102)'
+		);
+	});
+});
+
+test('Use the view more button to select a display page template from the select modal and change preview page', async ({
+	displayPageTemplatesPage,
+	page,
+	site,
+	styleBooksPage,
+}) => {
+	const displayPageTemplateNameBlogsEntry = getRandomString();
+	const displayPageTemplateNameCategory = getRandomString();
+	const displayPageTemplateNameDocuments = getRandomString();
+	const displayPageTemplateNameWebContent1 = getRandomString();
+	const displayPageTemplateNameWebContent2 = getRandomString();
+	const styleBookName = getRandomString();
+
+	await test.step('Create and publish multiple display page templates', async () => {
+		await displayPageTemplatesPage.goto(site.friendlyUrlPath);
+
+		await displayPageTemplatesPage.createTemplate({
+			contentSubtype: 'Basic Web Content',
+			contentType: 'Web Content Article',
+			name: displayPageTemplateNameWebContent1,
+		});
+
+		await displayPageTemplatesPage.createTemplate({
+			contentSubtype: 'Basic Web Content',
+			contentType: 'Web Content Article',
+			name: displayPageTemplateNameWebContent2,
+		});
+
+		await displayPageTemplatesPage.createTemplate({
+			contentType: 'Blogs Entry',
+			name: displayPageTemplateNameBlogsEntry,
+		});
+
+		await displayPageTemplatesPage.createTemplate({
+			contentSubtype: 'Basic Document',
+			contentType: 'Document',
+			name: displayPageTemplateNameDocuments,
+		});
+
+		await displayPageTemplatesPage.createTemplate({
+			contentType: 'Category',
+			name: displayPageTemplateNameCategory,
+		});
+	});
+
+	await test.step('Click on the view more button to select a new display page template from the modal', async () => {
+		await styleBooksPage.goto(site.friendlyUrlPath);
+
+		await styleBooksPage.create(styleBookName);
+
+		await clickAndExpectToBeVisible({
+			autoClick: true,
+			target: page.getByRole('button', {name: 'More'}),
+			trigger: page.getByRole('button', {
+				name: displayPageTemplateNameCategory,
+			}),
+		});
+
+		await page
+			.frameLocator('iframe[title="Select"]')
+			.getByRole('button', {
+				name: 'Select ' + displayPageTemplateNameWebContent1,
+			})
+			.click();
+
+		const loadingAnimation = page.locator('.loading-animation').first();
+
+		await loadingAnimation.waitFor();
+		await loadingAnimation.waitFor({state: 'hidden'});
+
+		expect(
+			page.getByRole('button', {name: displayPageTemplateNameWebContent1})
+		).toBeVisible();
```

Comment:
I see you wait for the loading animation to appear, disappear and then assert that the `displayPageTemplateNameWebContent1` is visible. It seems to me that `displayPageTemplateNameWebContent1` is only available when the loading animation is complete. If so, I think you can remove the lines regarding the loading animation and only wait for 
 `displayPageTemplateNameWebContent1`  to be visible. 

Playwright will check if the element is visible every `x` milliseconds, if not, it wait and try again.

Reply:
If what I just said doesn't apply, please just add a blank line after 238, order matters.